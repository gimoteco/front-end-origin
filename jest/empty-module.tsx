import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
