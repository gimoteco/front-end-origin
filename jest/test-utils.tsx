import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
