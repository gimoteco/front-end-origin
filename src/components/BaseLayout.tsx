import * as React from 'react';
import Fade from 'react-reveal/Fade';
import HouseIcon from '../icons/house.svg';
import { calculateMonthlyDeposit, formatMoney } from '../utils/format';
import { parseDateParts } from '../utils/parser';
import * as Style from './BaseLayout.styles';
import Button from './Button';
import Card from './Card';
import Field from './Field';
import Header from './Header';
import { InputType } from './Input';
import MonthlyValue from './MonthlyValue';
import Segment from './Segment';

export default function BaseLayout() {
  const [total, setTotal] = React.useState(null);
  const [reachDate, setReachDate] = React.useState(new Date(2020, 9, 31));
  const { numberOfDeposits, depositValue } = calculateMonthlyDeposit(
    total,
    reachDate
  );
  const { month, year } = parseDateParts(reachDate);

  return (
    <Style.BaseLayoutWrapper>
      <Header />
      <Style.MainContent>
        <Fade>
          <Style.Instruction>
            Let's plan your <strong>saving goal.</strong>
          </Style.Instruction>
        </Fade>

        <Card title="Buy a house" subtitle="Saving goal" icon={<HouseIcon />}>
          <form>
            <Style.FormFieldsWrapper>
              <Field
                id="totalAmount"
                label="Total amount"
                value={total}
                type={InputType.Money}
                onChange={(newValue: number) => setTotal(newValue)}
              />

              <Field
                id="reachDate"
                label="Reach goal by"
                value={reachDate}
                type={InputType.Month}
                onChange={(newValue: Date) => setReachDate(newValue)}
              />

              <Segment>
                {{
                  top: <MonthlyValue value={depositValue} />,
                  bottom: (
                    <p>
                      Youâ€™re planning{' '}
                      <strong>{numberOfDeposits} monthly deposits</strong> to
                      reach your <strong>{formatMoney(total)}</strong> goal by{' '}
                      <strong>
                        {month} {year}.
                      </strong>
                    </p>
                  )
                }}
              </Segment>
            </Style.FormFieldsWrapper>

            <Button>Confirm</Button>
          </form>
        </Card>
      </Style.MainContent>
    </Style.BaseLayoutWrapper>
  );
}
