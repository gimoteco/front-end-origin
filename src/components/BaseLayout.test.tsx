import React from 'react';
import { render } from '../../jest/test-utils';
import BaseLayout from './BaseLayout';

describe('BaseLayout', () => {
  it('should render according snapshot', () => {
    const { container } = render(<BaseLayout>Hello World</BaseLayout>);

    expect(container).toMatchSnapshot();
  });
});
