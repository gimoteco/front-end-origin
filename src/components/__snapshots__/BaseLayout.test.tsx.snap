// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout should render according snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa hFUtUI"
  >
    <header
      class="sc-htpNat fQvjWA"
    >
      <div
        class="react-reveal"
      >
        <span />
      </div>
    </header>
    <main
      class="sc-bwzfXH hEEQmz"
    >
      Hello World
    </main>
  </div>
</div>
`;
